packages/core/src/utils/react-query-utils.ts

```diff
@@ -114,6 +103,34 @@ export function invalidateQuery<TInput, TResult, T extends QueryFn>(
   return queryCache.invalidateQueries(queryKey)
 }
 
+export function setQueryData<TInput, TResult, T extends QueryFn>(
+  resolver: T | Resolver<TInput, TResult> | EnhancedResolverRpcClient<TInput, TResult>,
+  params: TInput,
+  newData: TResult | ((oldData: TResult | undefined) => TResult),
+  opts: MutateOptions = {refetch: true},
+): Promise<void | ReturnType<typeof queryCache.invalidateQueries>> {
+  if (typeof resolver === "undefined") {
+    throw new Error("setQueryData is missing the first argument - it must be a resolver function")
+  }
+  const queryKey = getQueryKey(resolver, params)
+
+  return new Promise((res) => {
+    queryCache.setQueryData(queryKey, newData)
+    let result: void | ReturnType<typeof queryCache.invalidateQueries>
+    if (opts.refetch) {
+      result = invalidateQuery(resolver, params)
+    }
+    if (isClient) {
+      // Fix for https://github.com/blitz-js/blitz/issues/1174
+      requestIdleCallback(() => {
```

Comment:
as shown in #1214 , requestIdleCallback is not supported in safari. Could you change this to use setTimeout maybe?

Reply:
Thanks for taking a look at this!

I believe this was addressed by https://github.com/blitz-js/blitz/pull/1247 it's using the shim, not `window.requestIdleCallback`.

Reply:
yeah, exactly. That shim should work here too, I believe

Reply:
Sorry, I'm a bit confused. It's already using that shim.

Reply:
oh, sorry. I didn't see the import and thought that this was implicitly using the function from "window". Thanks for clearing this up :)

Reply:
No worries!